cmd/subnetcmd/import_from_api.go

```diff
@@ -0,0 +1,282 @@
+				"Please provide an API URL of such a node so we can query its VM version (e.g. 111.22.33.44:5555)")
+			if err != nil {
+				return err
+			}
+			ctx, cancel := context.WithTimeout(context.Background(), constants.RequestTimeout)
+			defer cancel()
+			infoAPI := info.NewClient(nodeURL)
+			options := []rpc.Option{}
+			reply, err = infoAPI.GetNodeVersion(ctx, options...)
+			if err != nil {
+				return fmt.Errorf("failed to query node - is it running and reachable? %w", err)
+			}
+		}
+	}
+
+	var subnetID ids.ID
+	if subnetIDstr == "" {
+		subnetID, err = app.Prompt.CaptureID("What is the ID of the subnet?")
+		if err != nil {
+			return err
+		}
+	} else {
+		subnetID, err = ids.FromString(subnetIDstr)
+		if err != nil {
+			return err
+		}
+	}
+
+	var pubAPI string
+	switch network {
+	case models.Fuji:
+		pubAPI = constants.FujiAPIEndpoint
+	case models.Mainnet:
+		pubAPI = constants.MainnetAPIEndpoint
+	}
+	client := platformvm.NewClient(pubAPI)
+	ctx, cancel := context.WithTimeout(context.Background(), constants.RequestTimeout)
+	defer cancel()
+	options := []rpc.Option{}
+
+	ux.Logger.PrintToUser("Getting information from the %s network...", network.String())
+
+	chains, err := client.GetBlockchains(ctx, options...)
+	if err != nil {
+		return err
+	}
+
+	var (
+		blockchainID, vmID ids.ID
+		subnetName         string
+	)
+
+	for _, ch := range chains {
+		// NOTE: This supports only one chain per subnet
+		if ch.SubnetID == subnetID {
+			blockchainID = ch.ID
+			vmID = ch.VMID
+			subnetName = ch.Name
+			break
+		}
+	}
+
+	if blockchainID == ids.Empty || vmID == ids.Empty {
+		return fmt.Errorf("subnet ID %s not found on this network", subnetIDstr)
+	}
+
+	ux.Logger.PrintToUser("Retrieved information. BlockchainID: %s, Name: %s, VMID: %s",
+		blockchainID.String(),
+		subnetName,
+		vmID.String(),
+	)
+	// TODO: it's probably possible to deploy VMs with the same name on a public network
+	// In this case, an import could clash because the tool supports unique names only
+
+	genBytes, err := os.ReadFile(genesisFilePath)
+	if err != nil {
+		return err
+	}
+
+	if err = app.WriteGenesisFile(subnetName, genBytes); err != nil {
+		return err
+	}
+
+	vmType := getVMFromFlag()
+	if vmType == "" {
+		subnetTypeStr, err := app.Prompt.CaptureList(
+			"What's this VM's type?",
+			[]string{models.SubnetEvm, models.SpacesVM, models.CustomVM},
+		)
+		if err != nil {
+			return err
+		}
+		vmType = models.VMTypeFromString(subnetTypeStr)
+	}
+
+	sc := &models.Sidecar{
+		Name:            subnetName,
+		VM:              vmType,
+		ImportedFromAPM: false,
+		Networks: map[string]models.NetworkData{
+			network.String(): {
+				SubnetID:     subnetID,
+				BlockchainID: blockchainID,
+			},
+		},
+		Subnet:  subnetName,
+		Version: constants.SidecarVersion,
+	}
+
+	var versions []string
+	switch vmType {
+	case models.SubnetEvm:
+		versions, err = app.Downloader.GetAllReleasesForRepo(constants.AvaLabsOrg, constants.SubnetEVMRepoName)
```

Comment:
This is a serious question. I'm not implying the answer is yes. Is this a better approach than just asking the user to enter the version with text? That's currently what we do in the creation flow.

Reply:
If we decide yes, then we should add this into the creation flow.

Reply:
My opinion is that asking the user could result in more uncertainty. We also have the general pattern to make things for the user as easy as possible.

Reply:
Also I think that the creation flow is somewhat different in that the user is probably indeed creating the subnet for the first time, and might be in a different mind (and time) context than when importing a subnet from a running network. The latter argument also gets strengthened by the fact that this probably stems from having created a subnet via subnet-cli, which indeed is a different context altogether than our cli.

Reply:
Keep the prompting here but also replace it in the creation flow